modules/helfi_etusivu_entities/src/Plugin/Block/SurveyBlock.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\helfi_etusivu_entities\Plugin\Block;
+
+use Drupal\Component\Utility\Xss;
+use Drupal\Core\Block\Attribute\Block;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Utility\Error;
+use Drupal\helfi_etusivu_entities\Plugin\ExternalEntities\StorageClient\Surveys;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+
+/**
+ * Provides a 'Surveys' block.
+ */
+#[Block(
+  id: "surveys",
+  admin_label: new TranslatableMarkup("Surveys"),
+)]
+final class SurveyBlock extends EtusivuEntityBlockBase {
+
+  /**
+   * Entity types supported by survey and the corresponding field names.
+   */
+  private const ENTITY_TYPE_FIELDS = [
+    'node' => 'field_survey_content_pages',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    $currentEntity = $this->getCurrentPageEntity(array_keys(self::ENTITY_TYPE_FIELDS));
+
+    if ($survey = $this->getSurvey($currentEntity)) {
```

Comment:
Since this only displays one survey, should we prioritize global or local surveys?

Reply:
The priorization logic is: most recent survey first.